corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -219,6 +224,10 @@ hqDefine("data_dictionary/js/data_dictionary", [
                     }
 
                     _.each(group.properties(), function (element, index) {
+                        if (element.deleted && !element.id) {
```

Comment:
I believe if we mark even new ones as deleted like suggested [here](https://github.com/dimagi/commcare-hq/pull/34073/files#r1474504392) we would not need to add `id` at all and just check for `deleted`.

Reply:
We would still need to check for `id` here since we do not want to try and persist unsaved new case properties which have been immediately deleted (has no id).

Reply:
won't these new ones have deleted marked as true if we do what the other comment is saying?

Reply:
It is true that both new and existing case properties will have `deleted` marked as true. What makes it challenging however, is that there is no way to differentiate which case properties are persisted in the DB and which aren't (as we don't want to send non-persisted deleted props to the back-end). To avoid doing this, we use the ID to check which props exist in the DB and simply don't send deleted props that haven't been saved to the DB yet.

TL:DR - Checking the ID helps determine which delete requests should be sent to the back-end.

Reply:
Correct me if I am wrong,
any deleted case property is not sent back to HQ on save?
I thought this was true and so any new or existing case property won't be sent to HQ if it was marked as deleted.

Reply:
> any deleted case property is not sent back to HQ on save?

We do send deleted case properties back to HQ on save. I think the important thing to note is that case properties are simply marked for delete. Since already saved case properties are saved in the DB, we need to send these case properties that are marked for delete back to HQ so that it can remove these properties from the DB.

Reply:
Before the ID check we would send all case properties marked for delete to the back-end, regardless of whether it actually exists in the DB or not. This resulted in the back-end trying to delete a case property that doesn't exist and ultimately causing an exception to be returned to the UI.

Reply:
Apologies, I meant for the case properties that were present on HQ. How does HQ know which one to delete?

Reply:
For case properties we have an already existing function for updating the case prop with any new changes. This function was extended to check whether the `deleted` flag is true and, if so, remove the case prop instead of updating it.

For more context, you can have a look at [this commit](https://github.com/dimagi/commcare-hq/pull/33956/commits/ace35f75811b69f1daffaa8c6d57aebf95d0e2a0) which adds in the back-end implementation for deleting case types/props.